File: Intervals/435. Non-overlapping Intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        if not intervals:
            return 0 
        res = 0
        intervals.sort(key = lambda x : x[0])
        
        prevEnd = intervals[0][1]
        
        for i in range(1, len(intervals)):
            currentStart, currentEnd = intervals[i]
            
            if currentStart < prevEnd: # if intersect
                res += 1 
                prevEnd = min(prevEnd, currentEnd)
            else:
                prevEnd = currentEnd
                
        return res

File: Arrays/435. Non-overlapping Intervals.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());
        int prev = -1e9;
        int res = 0;
        for (const auto& interval: intervals) {
            int a = interval[0], b = interval[1];
            if (a < prev) {
                res++, prev = min(prev, b);
            } else {
                prev = max(prev, b);
            }

        }

        return res;
    }
};